packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -201,6 +204,32 @@ export class SelectedNetworkController extends BaseController<
         });
       },
     );
+
+    onPreferencesStateChange(({ useRequestQueue }) => {
+      if (this.#useRequestQueuePreference !== useRequestQueue) {
+        if (!useRequestQueue) {
+          Object.keys(this.state.domains).forEach((domain) => {
+            this.#unsetNetworkClientIdForDomain(domain);
+          });
+        } else {
+          this.#domainProxyMap.forEach((_: NetworkProxy, domain: string) => {
+            const { selectedNetworkClientId } = this.messagingSystem.call(
+              'NetworkController:getState',
+            );
+            // can't use public setNetworkClientIdForDomain because it will throw an error
+            // rather than simply skip if the domain doesn't have permissions which can happen
+            // in this case since proxies are added for each site the user visits
+            if (this.#domainHasPermissions(domain)) {
+              this.#setNetworkClientIdForDomain(
+                domain,
+                selectedNetworkClientId,
+              );
+            }
+          });
```

Comment:
thoughts on moving this block of logic into a private method?

Reply:
We could. Happy to do it. Don't feel strongly about it either way!

Reply:
done here: https://github.com/MetaMask/core/pull/4130/commits/37fb0106594ba5664ce3ad33c8d2879caa5bbf32